### PR TITLE
switch from building with distutils to setuptools in setup.py

### PR DESCRIPTION
**Pull request by dkeeney**
I wanted to see if everything builds in CI with this change.  But do not merge until we here from @be4pro who I requested to check if it works when only Visual Studio 2019 is installed.

### CODE REVIEW

**Comment by dkeeney** [comment]
It appears that the build is broken at the moment.
This PR is building basically what is in master and it fails.
I will assume that @breznak is working on something...  No problem,  I will try later.

**Comment by breznak** [comment]
> I will assume that @breznak is working on something... No problem, I will try later.
Merge state

 I havent been touching the repo at that time.

**Comment by dkeeney** [comment]
This PR has only a couple of comment changes.  So it should build the same as the master.

**Comment by dkeeney** [comment]
It seems to fail on "python setup.py configure"
I think this is redundant since "python setup.py install" does it all.

**Comment by breznak** [comment]
> This PR has only a couple of comment changes. So it should build the same as the master.

that is weird, on other PRs master is passing. 
Agree these changes shouldn't affect anything

**Comment by dkeeney** [comment]
My other PR #726 is also failing in the same way.

**Comment by breznak** [comment]
> My other PR #726 is also failing in the same way.

if you see recently merged PRs, all were passing OK. 
So what could this be? 
- CI changed/updated something
- some of our deps that are not fixed to version, but use "master" changed silently
- other ideas...??

**Comment by breznak** [comment]
> It seems to fail on "python setup.py configure"
> > some of our deps that are not fixed to version, but use "master" changed silently 

there we go..

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -51,14 +51,12 @@ jobs:
       run: |
         sudo apt-get update
         sudo apt-get -y install gcc-8 g++-8
-        python -m pip install --upgrade pip setuptools wheel
```
Comment (breznak) [inline .github/workflows/build.yml]: FYI @dkeeney added the setuptools>=34.4.0 to requirements and removed the manual upgrade from CI (let's see if other pip,wheel break and need specific version too)

**Comment by breznak** [comment]
Either it was an unrelated fuckup, or this change fixes that..anyway :+1: 
Please review and merge, @dkeeney

**Comment by dkeeney** [comment]
I am glad you found it.  I had no idea where to look for this crash.
Sorry I assumed that it was something you were doing. I hope you did not take that wrong. I did not intend to imply that you messed up the master.

**Comment by breznak** [comment]
> to imply that you messed up the master.

no worries, I didnt :smiley: TBO, I still do not know if/what made the master failing..

**Pull request merged**